src/settings.js

```diff
@@ -0,0 +1,567 @@
+        var widgetPrefix = that.options.widgets.gradePrefix,
+            widgetGrade,
+            widgetGradesToTypes = that.options.widgets.typesToGrades;
+
+        if (!widgetGradesToTypes) {
+            console.log("`widgetGradesToTypes` missing - Widget grades to types not supplied.");
+        }
+
+        widgetGrade = widgetGradesToTypes[type];
+        if (!widgetGrade) {
+            console.log("Widget " + type + " is not supported.");
+        }
+
+        return widgetPrefix + "." + widgetGrade;
+    };
+
+    // TODO simplify - probably do this sort of mapping somewhere up the chain?
+    gpii.app.settings.settingsVisualizer.getRequiredResources = function (settingsVisualizer, settings) {
+        function appendWidgetResources(resources, settings, settingsVisualizer) {
+            var widgetGrades = settings
+                .map(function (setting) { return setting.type; })
+                .filter(function uniq(setting, idx, settings) { return settings.indexOf(setting) === idx; })
+                .map(
+                    // receives the GPII setting type
+                    gpii.app.settings.settingsVisualizer.getWidgetGrade.bind(null, settingsVisualizer)
+                );
+
+            widgetGrades.forEach(function (widgetGrade) {
+                var template = fluid.defaults(widgetGrade).template;
+                resources[widgetGrade] = template;
+            });
+        }
+
+        // TODO remove this ugliness
+        var resources = {
+            rowTemplate:  fluid.defaults("gpii.app.settings.settingRow").template
+        };
+
+        appendWidgetResources(resources, settings, settingsVisualizer);
+
+        return resources;
+    };
+
+
+    /**
+     * Responsible for visualizing the list of user settings.
+     * Behaviour:
+     *  - Loads templates for the setting row and widget that is to be used
+     *  - Once templates are loaded, dynamicaly creates and supplies container for a `settingRow` element, for every user setting
+     * Expects: list of settings
+     */
+    fluid.defaults("gpii.app.settings.settingsVisualizer", {
+        gradeNames: "fluid.viewComponent",
+        model: {
+            // TODO modelRelay - transform received settings with 
+            //  pcp specific data (widgets)
+            settings: []
+        },
+        widgets: {
+            // Represents a map for GPII settings schema type to PCP widget grade to be used
+            typesToGrades: {
+                array: "multipicker",
+                boolean: "switch",
+                string: "dropDown",
+                number: "slider",
+                numberStep: "stepper",
+                text: "textfield",
+                radio: "radioToggle"
+            },
+            gradePrefix: "gpii.app.settings.widgets"
+        },
+        markups: {
+            container: "<div class=\"%containerClass\"></div>",
+            containerClassPrefix: "flc-settingListRow-%id"
+        },
+        components: {
```

Comment:
What I recommend you do here is place a top-level component named "exemplars" with one child for each widget type.
E.g. something like this:
exemplars: {
    type: "gpii.pcp.widgetExemplars"
}

where that grade is:

fluid.defaults("gpii.pcp.widgetExemplars", {
     components: {
        multipicker: {
            type: "gpii.pcp.exemplars.multipicker"
        },
etc.

where for example 

fluid.defaults("gpii.pcp.exemplars.multipicker", {
    gradeNames: "gpii.pcp.exemplar",
    template: "dropDown.html",
    widgetGrade: "gpii.pcp.widgets.multipicker",
    schemaTypes: "array"
});

As a member you can then build up a map as follows:

members: {
    keyToExemplar: "@expand(gpii.pcp.indexExemplars({widgetExemplars})"
}

where indexExemplars is something like:

gpii.pcp.indexExemplars = function (exemplarRoot) {
    // Take a shallow clone of the component
    var togo = $.extend({}, exemplarRoot);
    // Remove all non-components from the clone
    return fluid.remove_if(togo, function (child) {
        return !fluid.isComponent(child);
    });
}

(this could be done in a more principled but long-winded way using fluid.queryIoCSelector and fluid.pathToComponent - but we simply agree with ourselves that all child components of the exemplar root are exemplars)

You could then convert this into a form suitable to be templates by then writing as a further member:

templates: "@expand(fluid.getMembers({that}.keyToExemplars, options.template)"

This map of template can then be sent directly to getRequiredResources - see above